### PR TITLE
Sample App Improvements

### PR DESCRIPTION
**Pull request by suchoX**
1. Added 'Get Chuck Norris Joke' API
2. Decreased default barricade response time to 150
3. Added proper UI for showing API response data

### CODE REVIEW

File: app/src/main/java/com/mutualmobile/barricade/sample/MainActivity.java [hunk 1]
```diff
@@ -100,8 +140,18 @@ private void init() {
...
-        .client(okHttpClient).addConverterFactory(GsonConverterFactory.create()).build();
+        .client(okHttpClient)
+        .addConverterFactory(GsonConverterFactory.create())
+        .build();
 
     gitHubApiService = retrofit.create(GitHubApiService.class);
   }
+
+  private void initRecyclerView() {
+
```
Comment (tushar-nallan) [inline app/src/main/java/com/mutualmobile/barricade/sample/MainActivity.java:151]: remove new line

File: app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+class RepoListAdapter extends RecyclerView.Adapter<RepoListAdapter.ViewHolder> {
+
+  private ArrayList<Repo> repoList;
+
+  RepoListAdapter(ArrayList<Repo> repoList) {
+    this.repoList = repoList;
+  }
+
+  void setRepoList(ArrayList<Repo> repoList) {
```
Comment (tushar-nallan) [inline app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java:22]: Take List as param

File: app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import com.mutualmobile.barricade.sample.api.model.Repo;
+import java.util.ArrayList;
+
+class RepoListAdapter extends RecyclerView.Adapter<RepoListAdapter.ViewHolder> {
+
+  private ArrayList<Repo> repoList;
```
Comment (tushar-nallan) [inline app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java:16]: Use List here

File: app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  RepoListAdapter(ArrayList<Repo> repoList) {
```
Comment (tushar-nallan) [inline app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java:18]: List param

File: app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    this.repoList = repoList;
+  }
+
+  void setRepoList(ArrayList<Repo> repoList) {
+    this.repoList = repoList;
+  }
+
+  @Override public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+    View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_repo, parent, false);
```
Comment (tushar-nallan) [inline app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java:27]: inline v

File: app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+  @Override public void onBindViewHolder(ViewHolder holder, int position) {
+    holder.repoTextView.setText(repoList.get(position).name);
+  }
+
+  @Override public int getItemCount() {
+    return repoList == null ? 0 : repoList.size();
+  }
+
+  class ViewHolder extends RecyclerView.ViewHolder {
```
Comment (tushar-nallan) [inline app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java:39]: make class static whenever possible

**Review by tushar-nallan — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java (hunk 6 above)
Comment (suchoX) [reply app/src/main/java/com/mutualmobile/barricade/sample/RepoListAdapter.java:39]: I am using `repoList` variable which belongs to the outer Adapter class in the inner class, so I don't think I can make the inner class static

**Pull request closed without merging**